rotkehlchen/globaldb/handler.py

```diff
@@ -2008,3 +2010,68 @@ def get_exchange_name_from_assetid(exchange: Location, asset_identifier: str) ->
+        """Returns a tuple with the mappings, their amount according to the filter_query, and total
+        amount without any filter. Mappings are in the form of a list of dicts with the keys,
+        asset, location, and location_symbol."""
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            mappings_total = cursor.execute(
+                'SELECT COUNT(*) FROM location_asset_mappings',
+            ).fetchone()[0]
+
+            query, bindings = filter_query.prepare(with_pagination=False)
+            mappings_count = cursor.execute(
+                f'SELECT COUNT(*) FROM location_asset_mappings {query}', bindings,
+            ).fetchone()[0]
+
+            query, bindings = filter_query.prepare()
+            cursor.execute(
+                f'SELECT local_id, location, exchange_symbol FROM location_asset_mappings {query}', bindings,  # noqa: E501
+            )
+            return [{
+                'asset': identifier,
+                'location': None if location is None else str(Location.deserialize_from_db(location)),  # noqa: E501
+                'location_symbol': symbol,
+            } for identifier, location, symbol in cursor], mappings_count, mappings_total
+
+    @staticmethod
+    def add_location_asset_mappings(entries: list[LocationAssetMappingUpdateEntry]) -> None:
+        """Adds the given mapping entries of asset identifiers and their symbols in the given
+        location to the location_asset_mappings table."""
+        with GlobalDBHandler().conn.write_ctx() as cursor:
+            cursor.executemany(
+                'INSERT INTO location_asset_mappings(local_id, location, exchange_symbol) VALUES(?, ?, ?)', [(  # noqa: E501
+                    entry.asset.serialize(),
+                    None if entry.location is None else entry.location.serialize_for_db(),
+                    entry.location_symbol,
+                ) for entry in entries],
+            )
+
+    @staticmethod
+    def update_location_asset_mappings(entries: list[LocationAssetMappingUpdateEntry]) -> None:
+        """Updates the mapped asset identifiers in the location_asset_mappings table against their
+        location symbol."""
+        with GlobalDBHandler().conn.write_ctx() as cursor:
+            cursor.executemany(
+                'UPDATE location_asset_mappings SET local_id=? WHERE location=? AND exchange_symbol=?', [(  # noqa: E501
+                    entry.asset.serialize(),
+                    None if entry.location is None else entry.location.serialize_for_db(),
+                    entry.location_symbol,
+                ) for entry in entries],
+            )
+
+    @staticmethod
+    def delete_location_asset_mappings(entries: list[LocationAssetMappingDeleteEntry]) -> None:
+        """Deletes the mappings of given asset identifiers in the given location from the
+        location_asset_mappings table."""
+        with GlobalDBHandler().conn.write_ctx() as cursor:
+            cursor.executemany(
```

Comment:
If a deletion of a non-existing entry is requested no error is shown/seen from what I see. Shouldn't it? Like we do in other apis? I think we count the cursor's change count.